### ISSUE TITLE
Does anybody at Blockchain.com still care for their company?

### ISSUE THREAD

**Issue by ghost** [blockchain/service-my-wallet-v3#459]
Well, I have to ask that one here since the customer support won't even bother to reply anywhere else - which is a result of the problem. Is anyone experiencing the same? To me, this whole company seems dead for years. Maybe it's due to the fact that they're interested in different stuff now (whatever that might be) or just the result of their job description including "unlimited free-to-choose freetime" - just kidding ;)

But this repository was most recently updated 1,5 years ago! What about security updates? What about bug fixing? There are trillions of issues that no one closes or at least responds to - and they are serious things. Your website tells some stuff, the rest users will have to guess - for example the exact limits. When we build a large-scale project and come across #371 , what are we gonna do while our huge user base is giving us their trust?

Our API key request contained specific, clear questions about API limits, costs and potential problems. What we got was an email sending us congrats for having successfully applied for an API key - and another email from an employee, if not a bot, telling us that we got an API key. When we replied and said that we wished to get a human response and answers to our questions and repeated our questions clearly visible, guess what we got... a seemingly automated message telling us "the problem you're facing can probably be solved by archiving addresses [... bla ...]". Even a dog could provide this level of support.

Is your API and user's security, problems, etc. such an unimportant thing to you - as well as your customer support is? What are you caring for instead if I may ask?

At this point, I suggest any company coming across this API **not to put trust in Blockchain.com** as it's not entirely clear how stable and secure your application based on their infrastructure is going to be.

Thank you for your attention. What are your experiences with this platform? Unfortunately this seems to be the only place I could get any feedback at all. Thanks again and have a nice day. Or as the friendly human Blockchain support uses to say:

Best,

-------------------------------------

P.S.: Don't close this issue before you make at least one commit to this repository or I'll call the FBI.

**Comment by ConKreter** [comment]
This is absolutely true. We are using blockchain.com a lot and it becomes more and more a pain. The new issue now since last week is that when you make an outgoing transfer the wallet sends it but doesnt recognize the "change" address. So it appears the wallet got emptied but in reality the funds are still there (which is easy to verify with electrum). So luckily no funds lost but API unusuable again. On the receiving side around 10,000 transaction per wallet it ALWAYS breaks with the timeout problem which was mentioned before. This is actually due to blockchain.com having a hard limit on cloudflare and since some months ago all communication runs through cloudflare even though we have an API key (which didnt used to be like this before).

Basically for us we had to incorporate a different wallet provider as a backup. This was the only way. A real shame but at the end of the day what is free is rarely good.

**Comment by ghost** [comment]
Thanks for sharing your experiences! I'm glad we haven't completely integrated the API into our project yet. Guess it's true that you shouldn't have huge expectations of free software (although there are many great free projects out there, no doubts). I didn't know about the Cloudflare thing - we use Cloudflare as well and although we don't know the technical details of Blockchain.com's infrastructure, it should be quite simple to either deactivate the Cloudflare restrictions for that particular api.blockchain.com subdomain or put special rules into place which deactivate limits for users who provide an API key.

Anyhow, as we're intending to proceed with our development, we're now also searching for another solution. The great thing about Blockchain.com's API is, with no doubt, its simplicity - while it is actually free. This is great since we won't have much funding. May I know which wallet provider you have used as a backup?

**Comment by vitalio-sh** [comment]
> This is absolutely true. We are using blockchain.com a lot and it becomes more and more a pain. The new issue now since last week is that when you make an outgoing transfer the wallet sends it but doesnt recognize the "change" address. So it appears the wallet got emptied but in reality the funds are still there (which is easy to verify with electrum). So luckily no funds lost but API unusuable again. On the receiving side around 10,000 transaction per wallet it ALWAYS breaks with the timeout problem which was mentioned before. This is actually due to blockchain.com having a hard limit on cloudflare and since some months ago all communication runs through cloudflare even though we have an API key (which didnt used to be like this before).
> 
> Basically for us we had to incorporate a different wallet provider as a backup. This was the only way. A real shame but at the end of the day what is free is rarely good.

I have similar problem:

I have been using this service for quite a long time and there have been no particular complaints (except for a bad API).



But very strange things happened today (the amounts of transaction will be rounded):

27 APRIL 2020

7 a.m. - on my wallet about 0.05 bitcoins. I make a transfer using API to 0.035 bitcoins. And instead of getting the change 0.015 bitcoin is sent to some unknown address.

This transaction was fully automatic and I didn't notice the 0.015 bitcoin missing. And I didn't see the problem until the 10:00 a.m. actions.



10 am - I charge my blockchain.com wallet with 0.45 Bitcoin to start the working day and make transfers to clients.

I need to make a transfer to 0.02 bitcoin. I do it through site directly blockchain.com - specify the recipient's address and enter 0.02 bitcoin. Clicking send and after a second I see that my wallet becomes 0 bitcoins. I open the transaction and see that 0.02 went to the right address, 0.43 went not to the change, but to an unknown address. I look at the transaction before that and see that at 7:00 a.m. the same thing happened.



I immediately write to tech support and describe the situation. I haven't heard back from them in six hours already...



12 o'clock in the afternoon - I make an expirement and refill my wallet with 0.013 bitcoin and transfer 0.012 bitcoin to my other wallet. And again - I do not receive change, and the amount of 0.001 Bitcoin is sent again to the same unknown wallet. And again, the interface of the site says that I allegedly indicated two recipients. The transaction was made through the site.



6 p.m. - I make one more extra and record video. I refill my wallet with 0.0006 bitcoin. I open the application on my phone with authorization in the same wallet (the phone is connected to a cellular network, not to wifi as on the computer). Send 0.0005 bitcoin to your wallet - this amount is gone right, and 0.0001 bitcoin is sent again to an unknown wallet.



In all 4 cases, the unknown wallet has the same address.

The support of blockchain.com is already informed. All addresses and transactions have been sent to them, as well as video proof that the problem is full on their side.



From blockchain.com I expect a response to me and the refund sent through their fault is not a purse that I did not have access



If necessary, I can post all data on transactions, wallets, screenshots and video evidence of this case.
